aten/src/ATen/native/Convolution.cpp

```diff
@@ -618,8 +635,9 @@ at::Tensor _convolution(
     }
 #endif
   } else if (input.device().type() == c10::DeviceType::CPU || input.device().type() == c10::DeviceType::CUDA) {
-    // TH/native only covers CPU/CUDA implementation.
-    if (params.groups == 1) {
+    if (params.use_caffe2_depthwise(input, weight)) {
+      output = depthwise3x3ConvNCHW(input, weight, bias, params.padding, params.stride, params.groups);
+    } else if (params.groups == 1) {
       output = at::_convolution_nogroup(
           input, weight, bias, params.stride, params.padding, params.dilation, params.transposed, params.output_padding);
     } else {
```

Comment:
I think Caffe 2's implementation is a super-set of NNPACK.  If that is the case I should be able to remove the else if and else code paths.